src/benchmark/operators/sort_benchmark.cpp

```diff
@@ -1,22 +1,181 @@
+#include <expression/expression_functional.hpp>
 #include <memory>
 
 #include "benchmark/benchmark.h"
 
 #include "../micro_benchmark_basic_fixture.hpp"
+#include "operators/limit.hpp"
 #include "operators/sort.hpp"
 #include "operators/table_wrapper.hpp"
+#include "synthetic_table_generator.hpp"
+#include "operators/join_hash.hpp"
 
 namespace opossum {
 
-BENCHMARK_F(MicroBenchmarkBasicFixture, BM_Sort)(benchmark::State& state) {
-  _clear_cache();
+class SortBenchmark : public MicroBenchmarkBasicFixture {
+ public:
+  void BM_Sort(benchmark::State& state) {
+    _clear_cache();
 
-  auto warm_up = std::make_shared<Sort>(_table_wrapper_a, ColumnID{0} /* "a" */, OrderByMode::Ascending);
-  warm_up->execute();
-  for (auto _ : state) {
-    auto sort = std::make_shared<Sort>(_table_wrapper_a, ColumnID{0} /* "a" */, OrderByMode::Ascending);
-    sort->execute();
+    auto warm_up = std::make_shared<Sort>(_table_wrapper_a, ColumnID{0} /* "a" */, OrderByMode::Ascending);
+    warm_up->execute();
+    for (auto _ : state) {
+      auto sort = std::make_shared<Sort>(_table_wrapper_a, ColumnID{0} /* "a" */, OrderByMode::Ascending);
+      sort->execute();
+    }
   }
-}
+
+ protected:
+  void GenerateCustomTable(const ChunkOffset& chunk_size, const size_t& row_count,
+                           const DataType data_type = DataType::Int) {
+    const auto table_generator = std::make_shared<SyntheticTableGenerator>();
+    const int num_columns = 1;
+    const int max_different_value = 10'000;
+    const std::vector<DataType> column_data_types = {num_columns, data_type};
+
+    // We only set _table_wrapper_a because this is the only table (currently) used in the Sort Benchmarks
+    _table_wrapper_a = std::make_shared<TableWrapper>(table_generator->generate_table(
+        {num_columns, {ColumnDataDistribution::make_uniform_config(0.0, max_different_value)}}, column_data_types,
+        row_count, chunk_size, std::vector<SegmentEncodingSpec>(num_columns, {EncodingType::Unencoded}), std::nullopt,
+        UseMvcc::No));
+    _table_wrapper_a->execute();
+  }
```

Comment:
All in all good, but I am wondering, whether we could reduce redunancy below even more by using the means for benchmark parameterization mentioned in the first PR.

Reply:
:+1: We've created a separate follow-up issue for that (#5) and will look into it latest when we implement the next micro benchmarks involving parameterisation.